src/luaclang.c

```diff
@@ -1,14 +1,367 @@
+        int diagnostics = lua_toboolean(L, 2);
+        CXIndex *idx = new_CXIndex(L);
+        *idx = clang_createIndex(exclude_pch, diagnostics);
+        return 1;
+}
+
+static luaL_Reg clang_function[] = {
+        {"createIndex", create_CXIndex},
+        {NULL, NULL}
+};
+
+
+/* --Index functions-- */
+
+/*      
+        Format - idx:disposeIndex()
+        Parameter -  idx - Index to be destroyed       
+        More info - https://clang.llvm.org/doxygen/group__CINDEX.html#ga166ab73b14be73cbdcae14d62dbab22a
+*/
+static int dispose_CXIndex(lua_State *L) 
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        luaL_argcheck(L, idx != NULL, 1, "arg disposed");
+        clang_disposeIndex(idx);
+        idx = NULL;
+        return 0;
+}
+
+/*      
+        Format - idx:parseTU(file_name)
+        Parameters - 1. idx - The index object with which the translation unit will be associated  
+                     2. file_name - The name of the source file to load    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#ga2baf83f8c3299788234c8bce55e4472e
+*/
+static int parse_TU(lua_State *L)
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        const char *file_name = lua_tostring(L, 2);
+        const char *args[] = {file_name};
+        CXTranslationUnit *tu = new_CXTU(L);
+        *tu = clang_parseTranslationUnit(idx, 0, args, 1, 0, 0, CXTranslationUnit_None);
+        return 1;
+}
+
+
+static luaL_Reg index_functions[] = {
+        {"disposeIndex", dispose_CXIndex},
+        {"parseTU", parse_TU},
+        {"__gc", dispose_CXIndex},
+        {NULL, NULL}
+};
+
+
+/* --Translation unit functions-- */
+
+/*      
+        Format - tu:disposeTU()
+        Parameter - tu - Translation unit to be destroyed       
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#gaee753cb0036ca4ab59e48e3dff5f530a
+*/
+static int dispose_CXTU(lua_State *L) 
+{
+        CXTranslationUnit tu = to_CXTU(L, 1);
+        luaL_argcheck(L, tu != NULL, 1, "arg disposed");
+        clang_disposeTranslationUnit(tu);
+        tu = NULL;
+        return 0;
+}
+
+/*      
+        Format - tu:getTUCursor()
+        Parameter - tu - Translation unit of which the cursor represents     
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__MANIP.html#gaec6e69127920785e74e4a517423f4391
+*/
+static int get_TU_cursor(lua_State *L) 
+{
+        CXTranslationUnit tu = to_CXTU(L, 1);
+        CXCursor* cur = new_CXCursor(L);
+        *cur = clang_getTranslationUnitCursor(tu);
+        if (clang_Cursor_isNull(*cur)) {
+                lua_pushnil(L);
+        }
+        return 1;
+}
+
+
+static luaL_Reg tu_functions[] = {
+        {"disposeTU", dispose_CXTU},
+        {"getTUCursor", get_TU_cursor},
+        {"__gc", dispose_CXTU},
+        {NULL, NULL}
 };
 
-int luaopen_luaclang (lua_State *L)
+
+/* --Cursor functions-- */
+
+/*      
+        Format - cur:getCursorSpelling()
```

Comment:
The description of the return value is missing from the documentation. Please review all of them.